### PR TITLE
Switch to theano-pymc.

### LINKED ISSUE #4137: Make PyMC3 use our Theano fork
We have forked Theano here: https://github.com/pymc-devs/theano-pymc

It's time that we switched. There's already a release on conda-forge.

### PR DESCRIPTION
**Pull request by twiecki**
Closes #4137.

Depending on what your PR does, here are a few things you might want to address in the description:
+ [x] what are the (breaking) changes that this PR makes?
+ [x] important background, or details about the implementation
+ [x] are the changes—especially new features—covered by tests and docstrings?
+ [x] consider adding/updating relevant example notebooks
+ [x] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

**Review by brandonwillard — APPROVED** [review]

**Comment by brandonwillard** [comment]
Oh, let's not forget that we need to get the GPU testing in order on our fork!

**Comment by AlexAndorra** [comment]
Should we merge this, or wait for GPU testing on Theano-PyMC?

**Comment by twiecki** [comment]
I think we should merge. GPU is not officially supported anyway.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## PyMC3 3.9.x (on deck)
 
 ### Maintenance
+- Switch the dependency of Theano to our own fork Theano-PyMC.
```
Comment (AlexAndorra) [inline RELEASE-NOTES.md]:
```suggestion
- Switch the dependency of Theano to our own fork, [Theano-PyMC](https://github.com/pymc-devs/Theano-PyMC).
```

**Review by AlexAndorra — APPROVED** [review]
Just added the link to our Theano fork -- you can just accept the suggestion and I'll merge 😉

**Pull request merged**